packages/clay-shared/src/useFocusManagement.ts

```diff
@@ -232,6 +271,19 @@ export function useFocusManagement(scope: React.RefObject<null | HTMLElement>) {
 				// when have React Portal.
 				if (nextElement) {
 					setInteractionObserver(nextElement);
+				} else {
+					// When the next element is not in scope, we check for the next
+					// focusable element in the document that is not in the React tree.
+					// This is particularly used for Portals.
+					const docElements = collectDocumentFocusableElements();
+
+					const index = docElements.indexOf(elements[0]);
```

Comment:
@bryceosterhaus for cases where we can have a lot of elements in the scope, wouldn't that break? it wouldn't have to be `docElements.indexOf(elements[position]);`.

Maybe I'm understanding this case, probably the other elements are the DropDown buttons that are inside a Portal, if you take the last one you will not find the closest one to it, because it is in the body and it would have to be the DropDown trigger button. I think we would have to test this more in other use cases, considering more elements on the screen.

Anyway, I fell into the same challenge when it was developed, oriented towards DOM, the Portal was always the problem, I had to make some inferences to try to get the element that was not in the Portal and then get the next element from it.

One element to test this is the Pagination Bar, which maybe you can find some cases.